### DISCUSSION TITLE
Default field values should apply to existing elements when added to a field layout

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/cms#8504] [Ideas]
When a field is added to a field layout, it should have a chance to apply its default value to all the elements that have that field layout.

For example, if you create a Checkboxes field with a default checkbox option, and then you add it to a category group’s field layout, all existing categories in that group should be updated immediately with the default option’s value.

**Comment by brandonkelly** [comment]
Would resolve #2445, and any similar issues we’ve had due to current behavior not being expected.

**Comment by vonscriptor** [comment]
This enhancement would also need to accommodate multi-site / translatable fields.

**Comment by ajp** [comment]
@brandonkelly Am I correct in seeing that this is now a 4.0 milestone?
So, in the event of adding a new field to an existing layout/entry type/etc, I'll need to manually re-save those entries to get the default value of that field. I tried using the commandline resave/entries, but it did not affect the existing entries.

**Comment by brandonkelly** [comment]
@ajp Currently there’s no built-in way to apply the default value to existing elements. The one exception is Dropdown fields, and only when saved via the CP, and that’s only because `<select>` tags _must_ have an option selected, so for the purposes of rendering the `<select>` HTML, the Dropdown field type will preselect the default option if there is no current value.

**Comment by angieherrera** [comment]
@brandonkelly is this still on the radar? If so, I'd say having an option to apply (or not) to already existing fields would be helpful, even if it's just a command on the CLI.

**Reply by brandonkelly** [reply]
No progress because there’s no clean way to communicate the need to update existing entries, to your other environments. It’s already possible with the CLI though!

```sh
php craft resave/entries --section=mySectionHandle --set=myFieldHandle --to=myDefaultValue
```

(See [Bulk-resaving elements](https://craftcms.com/knowledge-base/bulk-resaving-elements) for the full docs on how to use those `--set` / `--to` options.)
